code/frameworks/nextjs/src/nextImport/webpack.ts

```diff
@@ -0,0 +1,29 @@
+import type { Configuration as WebpackConfig } from 'webpack';
+import semver from 'semver';
+import { IgnorePlugin } from 'webpack';
+import { getNextjsVersion } from '../utils';
+
+export function configureNextImport(baseConfig: WebpackConfig) {
+  const nextJSVersion = getNextjsVersion();
+
+  const isNext12 = semver.satisfies(nextJSVersion, '~12');
+  const isNext13 = semver.satisfies(nextJSVersion, '~13');
+
+  baseConfig.plugins = baseConfig.plugins ?? [];
+
+  if (!isNext13) {
```

Comment:
should the checks be specific to the versions? because if users are using Next.js 11, then `!isNext13 => true, !isNext12 => true`, and soon Next.js 14 will fall under the same scenario.

Reply:
I don't understand the question fully.

So the package `next/legacy/image` is available in Next.js v13. Very likely, it will not be available in Next.js v14, because they very likely drop it. It was only introduced to make an upgrade from v12 to v13 as easy as possible. That means, that except version v13 `next/legacy/image` is not available, and therefore the `require` statements have to be ignored. 

The same is true for the `!isNext12` condition at the bottom. The `next/future/image` component is only available in Next12. It isn't available in other packages and therefore it has to be ignored in Next.js version !== 12.

Reply:
> should the checks be specific to the versions? because if users are using Next.js 11, then !isNext13 => true, !isNext12 => true, and soon Next.js 14 will fall under the same scenario.

And that's what we want. For version v11 and v14 these imports are not available and therefore they have to be ignored.